lib/codegen/meta/isa/x86/encodings.py

```diff
@@ -204,6 +205,14 @@ def enc_i32_i64_ld_st(inst, w_bit, recipe, *args, **kwargs):
 # bool constants.
 enc_both(base.bconst.b1, r.pu_id_bool, 0xb8)
 
+# floating-point constants equal to 0.0 can be encoded using `xorps`.
+# X86_32.enc(base.iconst.i32, *r.fimm_z(0x57),
+#             instp=IsZero(UnaryImm.imm))
+X86_64.enc(base.f32const, *r.f32imm_z(0x57),
+            instp=IsZero32BitFloat(UnaryIeee32.imm))
+X86_64.enc(base.f64const, *r.f64imm_z(0x57),
+            instp=IsZero64BitFloat(UnaryIeee64.imm))
```

Comment:
This looks good. Two notes:
 - It looks like this is missing the `0x0f` byte of the xorps instruction.
 - We'll also need to add `rex` forms for `X86_64`, to support xmm registers xmm8 through xmm15. There are other examples of adding `.rex(...)` calls in this file. Note that the `rex` form needs to be declared before the non-`rex` form.

Reply:
The recipe uses a `*args`, so you can pass the 0x0f as a separate argument, like `*r.f32imm_z(0x0f, 0x57)`